geometry/eolearn/geometry/transformations.py

```diff
@@ -0,0 +1,340 @@
+            warnings.warn('Given vector polygons contain some 3D geometries, they will be projected to 2D',
+                          RuntimeWarning)
+            vector_data.geometry = vector_data.geometry.apply(lambda geo: shapely.wkt.loads(geo.to_wkt()))
+
+        if self.values_column is None:
+            return zip(vector_data.geometry, [self.values] * len(vector_data.index))
+
+        if self.values is not None:
+            values = [self.values] if isinstance(self.values, (int, float)) else self.values
+            vector_data = vector_data[vector_data[self.values_column].isin(values)]
+
+        return zip(vector_data.geometry, vector_data[self.values_column])
+
+    def _get_raster_shape(self, eopatch):
+        """ Determines the shape of new raster feature, returns a pair (height, width)
+        """
+        if isinstance(self.raster_shape, (tuple, list)) and len(self.raster_shape) == 2:
+            if isinstance(self.raster_shape[0], int) and isinstance(self.raster_shape[1], int):
+                return self.raster_shape
+
+            feature_type, feature_name = next(self._parse_features(self.raster_shape)(eopatch))
+            return eopatch.get_spatial_dimension(feature_type, feature_name)
+
+        if self.raster_resolution:
+            resolution = float(self.raster_resolution.strip('m')) if isinstance(self.raster_resolution, str) else \
+                self.raster_resolution
+            width, height = bbox_to_dimensions(eopatch.bbox, resolution)
+            return height, width
+
+        raise ValueError('Could not determine shape of the raster image')
+
+    def _get_raster(self, eopatch, height, width):
+        """ Provides raster in which data will be written
+        """
+        feature_type, feature_name = self.raster_feature
+
+        if self.write_to_existing and feature_name in eopatch[feature_type]:
+            raster = eopatch[feature_type][feature_name].squeeze(axis=-1)
+
+            if (height, width) != raster.shape[:2]:
+                warnings.warn('Writing to existing raster with spatial dimensions {}, but dimensions {} were expected'
+                              ''.format(raster.shape[:2], (height, width)), RuntimeWarning)
+
+            return raster
+
+        return np.ones((height, width), dtype=self.raster_dtype) * self.no_data_value
+
+    def execute(self, eopatch):
+        """ Execute method
+
+        :param eopatch: input EOPatch
+        :type eopatch: EOPatch
+        :return: New EOPatch with vector data transformed into a raster feature
+        :rtype: EOPatch
+        """
+        if eopatch.bbox is None:
+            raise ValueError('EOPatch has to have a bounding box')
+
+        rasterization_shapes = self._get_rasterization_shapes(eopatch)
+        if not rasterization_shapes:
+            return eopatch
+
+        height, width = self._get_raster_shape(eopatch)
+        data_transform = rasterio.transform.from_bounds(*eopatch.bbox, width=width, height=height)
```

Comment:
If I understand correctly, the `data_transform` here is the transformation function?

Reply:
Not a function but an `Affine` transformation vector object that rasterio returns for given bounding box coordinates and dimensions. Maybe it would be better if it would be named `affine_transform`?